src/components/LeaderBoard.jsx

```diff
@@ -0,0 +1,249 @@
+
+class LeaderBoardItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showDetails: false,
+      itemType: null, // Delegated ro Directly donated
+      hasHistory: this.props.d.status === Donation.PAID, // PAID donations always has history
+    };
+
+    this.toggleDetail = this.toggleDetail.bind(this);
+    this.setItemType = this.setItemType.bind(this);
+    this.setItemHasHistory = this.setItemHasHistory.bind(this);
+  }
+
+  setItemType(type) {
+    this.setState({
+      itemType: type,
+    });
+  }
+
+  setItemHasHistory(hasHistory) {
+    this.setState({
+      hasHistory,
+    });
+  }
+
+  toggleDetail() {
+    this.setState(prevState => ({
+      ...prevState,
+      showDetails: !prevState.showDetails,
+    }));
+  }
+
+  render() {
+    const { d, rank, useAmountRemaining } = this.props;
+    const { donations, totalAmount, giver } = d;
+    return (
+      <Fragment>
+        <tr key={d._id}>
+          <td>
+            <button type="button" className="btn btn-info btn-sm" onClick={this.toggleDetail}>
+              <i className={this.state.showDetails ? 'fa fa-minus' : 'fa fa-plus'} />
+            </button>
+          </td>
+          <td className="font-weight-bold">{rank}</td>
+          <td className="td-user">
+            {giver && (
+              <Link to={`/profile/${giver.address}`}>
+                <Avatar size={30} src={getUserAvatar(giver)} round />
+                <span> {getUserName(giver)}</span>
+              </Link>
+            )}
+          </td>
+          <td className="td-donations-amount font-weight-bold">${totalAmount}</td>
+          <td />
+          <td className="" />
+        </tr>
+        {donations.map(dd => {
+          const donation = new Donation(dd);
+          let typeLabel;
+          let etherScanLink = '';
+          const { etherscan, homeEtherscan } = config;
+          switch (this.state.itemType) {
+            case 'delegated':
+              typeLabel = (
+                <span className="badge badge-info">
+                  <i className="fa fa-random " />
+                  Delegated
+                </span>
+              );
+              break;
+            case 'direct':
+              typeLabel = (
+                <span className="badge badge-warning">
+                  <i className="fa fa-plug" />
+                  Direct
+                </span>
+              );
+              break;
+            default:
+              typeLabel = null;
+          }
+
+          if (d.status === Donation.PAID || this.state.itemType === 'delegated') {
+            etherScanLink = etherscan && d.txHash ? `${etherscan}tx/${d.txHash}` : '';
+          } else if (this.state.itemType === 'direct') {
+            etherScanLink =
+              homeEtherscan && d.homeTxHash ? `${homeEtherscan}tx/${d.homeTxHash}` : '';
+          }
+          return (
+            <tr key={donation._id} style={this.state.showDetails ? {} : { display: 'none' }}>
```

Comment:
display 'none' is not a good solution on react... you can render the row if showDetails is true.